src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2738,6 +2738,12 @@ private BoundExpression BindIsOperator(BinaryExpressionSyntax node, DiagnosticBa
                 return new BoundIsOperator(node, operand, typeExpression, Conversion.NoConversion, resultType, hasErrors: true);
             }
 
+            if (node.Right is IdentifierNameSyntax name && name.Identifier.Text == "_")
+            {
+                // PROTOTYPE(patterns2): Need compat council confirmation that this new warning in existing code is acceptable.
+                diagnostics.Add(ErrorCode.WRN_IsTypeNamedUnderscore, name.Location, alias ?? (Symbol)targetType);
```

Comment:
>alias [](start = 84, length = 5)

Please test the alias case if we're not already doing so.

Reply:
See `UnderscoreDeclaredAndDiscardPattern_04` in a new iteration.

---
In reply to: [178314642](https://github.com/dotnet/roslyn/pull/25779#discussion_r178314642) [](ancestors = 178314642)